### PR TITLE
JDBC sources: Fix unexcepcted long Integer value failure

### LINKED ISSUE #3840: Source MySQL: unsigned numeric values larger than signed integer size break connector
## Expected Behavior
I expect Airbyte syncs to succeed when replicating a MySQL table which contains an unsigned int column in which some of the values are larger than 2^32-1. 

Slack thread: https://airbytehq.slack.com/archives/C01MFR03D5W/p1622696364083700

## Current Behavior
The sync breaks as reported by a user. They sent the following logs: 

```
2021-06-03 05:13:13 ERROR (/tmp/workspace/59/2) LineGobbler(voidCall):69 - Exception in thread "main" java.lang.RuntimeException: java.sql.SQLDataException: Value '4176160650' is outside of valid range for type java.lang.Integer
```

## Suggestions for solution
If we can identify inside the connector if the column is unsigned then we should use `Long` for unsigned and `Integer` otherwise. 

If we can't identify if the column is unsigned then we should always use long. 

## Steps to Reproduce
1. Create a table in a mysql db with column unsigned int 
1. insert a row with value 3428724653
1. run a sync from airbyte to any destination
1. :bomb: 

## Severity of the bug for you
High -- preventing user from using Airbyte

## Additional context

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200419820358096) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by DoNotPanicUA**
Issue: #3840

## What
Fix an error when JDBC source produces Integer value bigger than max Java Integer value. 

## How
Implement handler which tries to convert into Integer first. My assumption is that at most cases the value will be in the range. 
If it fails, the handler tries to convert into the Long type. In all other negative cases it will replace value by `null`.

## Pre-merge Checklist
- [x] *Run integration tests*
- [x] *Publish Docker images*

## Recommended reading order
1. `JdbcUtils.java`

### CODE REVIEW

File: airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java [hunk 1]
```diff
@@ -151,6 +151,19 @@ private static void setJsonField(ResultSet r, int i, ObjectNode o) throws SQLExc
     }
   }
 
+  /**
+   * In some sources Integer might have value larger than {@link Integer#MAX_VALUE}. E.q. MySQL has
+   * unsigned Integer type, which can contain value 3428724653. If we fail to cast Integer value, we
+   * will try to cast Long.
+   */
+  private static void putInteger(ObjectNode node, String columnName, ResultSet r, int i) {
```
Comment (yaroslav-hrytsaienko) [inline airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:159]: @DoNotPanicUA please use more significant naming for r and i. Another point should the method really be static?

Comment (DoNotPanicUA) [reply airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java]: @yaroslav-hrytsaienko 
The names are fixes. 
About static - yes. it's utility class where all methods are static. As result, this method is been called only inside other static method.

**Review by yaroslav-hrytsaienko — APPROVED** [review]

**Review by masonwheeler — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
LGTM -- please publish and merge. Also, can you add this to the data type tests? 

BTW see the definition-of-done checklist here to help clarify what next steps would be: https://github.com/airbytehq/airbyte/pull/3864

**Comment by sherifnada** [comment]
@DoNotPanicUA also can you update the PR title to match best practices for PR titles?

**Comment by DoNotPanicUA** [comment]
> LGTM -- please publish and merge. Also, can you add this to the data type tests?

This case is added to the PR with tests. Corresponding [commit](https://github.com/airbytehq/airbyte/pull/3810/commits/562841f63a3f61ce2d76c7de0dd1b1e7f09715fe).

**Pull request merged**